tests/components/jewish_calendar/test_sensor.py

```diff
@@ -445,3 +445,56 @@ def test_issur_melacha_sensor(self, now, candle_lighting, havdalah,
                 sensor.async_update(),
                 self.hass.loop).result()
             assert sensor.state == result
+
+    omer_params = [
+        make_nyc_test_params(dt(2019, 4, 21, 0, 0), 1),
+        make_jerusalem_test_params(dt(2019, 4, 21, 0, 0), 1),
+        make_nyc_test_params(dt(2019, 4, 21, 23, 0), 2),
+        make_jerusalem_test_params(dt(2019, 4, 21, 23, 0), 2),
+        make_nyc_test_params(dt(2019, 5, 23, 0, 0), 33),
+        make_jerusalem_test_params(dt(2019, 5, 23, 0, 0), 33),
+        make_nyc_test_params(dt(2019, 6, 8, 0, 0), 49),
+        make_jerusalem_test_params(dt(2019, 6, 8, 0, 0), 49),
+        make_nyc_test_params(dt(2019, 6, 9, 0, 0), 0),
+        make_jerusalem_test_params(dt(2019, 6, 9, 0, 0), 0),
+        make_nyc_test_params(dt(2019, 1, 1, 0, 0), 0),
+        make_jerusalem_test_params(dt(2019, 1, 1, 0, 0), 0),
+    ]
+    omer_test_ids = [
+        "nyc_first_day_of_omer",
+        "israel_first_day_of_omer",
+        "nyc_first_day_of_omer_after_tzeit",
+        "israel_first_day_of_omer_after_tzeit",
+        "nyc_lag_baomer",
+        "israel_lag_baomer",
+        "nyc_last_day_of_omer",
+        "israel_last_day_of_omer",
+        "nyc_shavuot_no_omer",
+        "israel_shavuot_no_omer",
+        "nyc_jan_1st_no_omer",
+        "israel_jan_1st_no_omer",
+    ]
+
+    @pytest.mark.parametrize(["now", "candle_lighting", "havdalah", "diaspora",
+                              "tzname", "latitude", "longitude", "result"],
+                             omer_params, ids=omer_test_ids)
+    def test_omer_sensor(self, now, candle_lighting, havdalah,
+                         diaspora, tzname, latitude, longitude,
+                         result):
+        """Test Omer Count sensor output."""
+        time_zone = get_time_zone(tzname)
+        set_default_time_zone(time_zone)
+        test_time = time_zone.localize(now)
+        self.hass.config.latitude = latitude
+        self.hass.config.longitude = longitude
+        sensor = JewishCalSensor(
```

Comment:
The tests will be more robust if we set up the component and platform via `setup_component` or `async_setup_component`, let the platform add an entity, mock time passing to have an update then assert that the entity state is correct by getting it from the core state machine.

Reply:
Sounds like a great idea. Do you have an example for something of that type?

Reply:
Here's an example that does something similar and also calls a service:
https://github.com/home-assistant/home-assistant/blob/dev/tests/components/manual/test_alarm_control_panel.py#L75-L107